var/spack/repos/builtin/packages/octave/package.py

```diff
@@ -182,12 +184,19 @@ def configure_args(self):
                 "--with-java-includedir=%s" % spec['jdk'].prefix.include,
                 "--with-java-libdir=%s"     % spec['jdk'].prefix.lib
             ])
+        else:
+            config_args.append("--disable-java")
 
         if '~opengl' in spec:
             config_args.extend([
                 "--without-opengl",
                 "--without-framework-opengl"
             ])
+        else:
+            config_args.extend([
+                "--without-opengl",
+                "--without-framework-opengl"
```

Comment:
This one should be reversed. Can you change the first half of the if-statement to be `if '+opengl' in spec:` and flip the with/without?